e2etest/l2tests/assignment.go

```diff
@@ -678,4 +698,269 @@ var _ = ginkgo.Describe("IP Assignment", func() {
+			pool1 := metallbv1beta1.IPAddressPool{
+				ObjectMeta: metav1.ObjectMeta{Name: "test-ns-dualstack-pool"},
+				Spec: metallbv1beta1.IPAddressPoolSpec{
+					Addresses: []string{
+						v4PoolAddresses,
+					},
+					AllocateTo: &metallbv1beta1.ServiceAllocation{Namespaces: []string{testNamespace}},
+				},
+			}
+			resources := config.Resources{
+				Pools: []metallbv1beta1.IPAddressPool{pool1},
+			}
+
+			err := ConfigUpdater.Update(resources)
+			Expect(err).NotTo(HaveOccurred())
+
+			ginkgo.By("Creating the service with PreferDualStack policy")
+			svc1, _ := service.CreateWithBackend(cs, testNamespace, "svc-test-ns-pool-1", func(svc *v1.Service) {
+				svc.Spec.IPFamilyPolicy = ptr.To(v1.IPFamilyPolicyPreferDualStack)
+			})
+			Expect(err).NotTo(HaveOccurred())
+
+			ginkgo.By("Validating that 1 IP is assigned to the service")
+			Eventually(func() int {
+				svc1, err = cs.CoreV1().Services(svc1.Namespace).Get(context.Background(), svc1.Name, metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				return len(svc1.Status.LoadBalancer.Ingress)
+			}, 5*time.Minute, 1*time.Second).Should(Equal(1))
+			ginkgo.By("validate LoadBalancer IP is allocated from pool1")
+			err = config.ValidateIPInRange([]metallbv1beta1.IPAddressPool{pool1}, jigservice.GetIngressPoint(
+				&svc1.Status.LoadBalancer.Ingress[0]))
+			Expect(err).NotTo(HaveOccurred())
+			ginkgo.By("Updating svc1 to have RequireDualStack family policy")
+			err = updateServiceIPFamilyPolicy(cs, svc1.Namespace, svc1.Name, ptr.To(v1.IPFamilyPolicyRequireDualStack))
+			Expect(err).NotTo(HaveOccurred())
+
+			ginkgo.By("Validating that the service has no IP")
+			Eventually(func() int {
+				svc1, err = cs.CoreV1().Services(svc1.Namespace).Get(context.Background(), svc1.Name, metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				return len(svc1.Status.LoadBalancer.Ingress)
+			}, 5*time.Minute, 1*time.Second).Should(Equal(0))
+
+			ginkgo.By("Updating svc1 to have PreferDualStack family policy")
+			err = updateServiceIPFamilyPolicy(cs, svc1.Namespace, svc1.Name, ptr.To(v1.IPFamilyPolicyPreferDualStack))
+			Expect(err).NotTo(HaveOccurred())
+			ginkgo.By("Validating that 1 IP is assigned to the service")
+			Eventually(func() int {
+				svc1, err = cs.CoreV1().Services(svc1.Namespace).Get(context.Background(), svc1.Name, metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				return len(svc1.Status.LoadBalancer.Ingress)
+			}, 5*time.Minute, 1*time.Second).Should(Equal(1))
+			ginkgo.By("validate LoadBalancer IP is allocated from pool1")
+			err = config.ValidateIPInRange([]metallbv1beta1.IPAddressPool{pool1}, jigservice.GetIngressPoint(
+				&svc1.Status.LoadBalancer.Ingress[0]))
+			Expect(err).NotTo(HaveOccurred())
+		})
+		ginkgo.It("Svc assigned from single-stack pool should not jump to dual-stack pool when it's available", func() {
```

Comment:
Can we see what happens if we remove an IP from a dual stack service (primary / secondary)? What is the expected behavior that we want?

Reply:
idk. What do you think should be the expected behavior?

Reply:
Imo in this case the ips should be reassigned? In case we have a dualstack pool available?

Reply:
yeah I don't have a strong opinion. On one side, we are promising the ip of the service not to change, on the other we change the ip with a good one in case the old one goes away.

I'd say we reassign if a dual stack pool is available, but we keep it as it is (cleaning one ip) if no other dual stack pool is available. Would it make sense? @oribon thoughts?

Reply:
sounds good to me (re-allocating or keeping if possible)